server/py/services/api/crud/model_monitoring/model_endpoints.py

```diff
@@ -62,84 +65,77 @@ def create_model_endpoint(
+        # 4. create a monitoring feature set & update the model endpoint object with
+        # the feature_set_uri, features and labels
+        # Get labels from model object if not found in model endpoint object
+        features = []
+        attributes = {}
+        if model_obj:
             if not model_endpoint.spec.label_names and model_obj.spec.outputs:
                 model_label_names = [
                     mlrun.feature_store.api.norm_column_name(f.name)
                     for f in model_obj.spec.outputs
                 ]
                 model_endpoint.spec.label_names = model_label_names
-
-            # Get algorithm from model object if not found in model endpoint object
-            if not model_endpoint.spec.algorithm and model_obj.spec.algorithm:
-                model_endpoint.spec.algorithm = model_obj.spec.algorithm
+                attributes["label_names"] = model_label_names
 
             features = cls._get_features(
                 model=model_obj,
-                run_db=run_db,
+                run_db=framework.api.utils.get_run_db_instance(db_session),
                 project=model_endpoint.metadata.project,
             )
             model_endpoint.spec.feature_names = [feature.name for feature in features]
-            # Create monitoring feature set if monitoring found in model endpoint object
-            if (
-                model_endpoint.spec.monitoring_mode
-                == mlrun.common.schemas.model_monitoring.ModelMonitoringMode.enabled
-            ):
-                monitoring_feature_set = cls.create_monitoring_feature_set(
-                    features=features,
-                    model_endpoint=model_endpoint,
-                    db_session=db_session,
-                )
-                # Link model endpoint object to feature set URI
-                model_endpoint.status.monitoring_feature_set_uri = (
-                    monitoring_feature_set.uri
-                )
+            attributes["feature_names"] = model_endpoint.spec.feature_names
+
+        if (
+            model_endpoint.status.monitoring_mode
+            == mlrun.common.schemas.model_monitoring.ModelMonitoringMode.enabled
+        ):
+            monitoring_feature_set = cls.create_monitoring_feature_set(
+                features=features,
+                model_endpoint=model_endpoint,
+                db_session=db_session,
+            )
+            # Link model endpoint object to feature set URI
+            model_endpoint.spec.monitoring_feature_set_uri = monitoring_feature_set.uri
+            attributes["monitoring_feature_set_uri"] = monitoring_feature_set.uri
+            # Create model monitoring schedules file
+            ModelMonitoringSchedulesFile.from_model_endpoint(model_endpoint).create()
+            # Create model monitoring stats files:
+            cls._create_model_monitoring_stats_files(model_endpoint=model_endpoint)
```

Comment:
See above.
What do you think?
I think it can be helpful even without a model URI. E.g., monitoring a router in the future